api/pb/types.proto

```diff
@@ -14,10 +14,14 @@ message Pin {
   repeated bytes Allocations = 3;
   sint32 MaxDepth = 4;
   bytes Reference = 5;
-  reserved 6,7;
```

Comment:
Should we be removing the reserved fields?

Reply:
why not.. they were reserved as a sort of separator from the fields that corresponded to PinOptions, but now I don't see reason to reserve them.

Reply:
I still think it would be a good idea to reserve some fields in case we add anything to the `Pin` struct itself, otherwise it will go after all the `PinOption` fields

Reply:
This will make pin options field take an extra byte for encoding. In my view it is not horrible that any new fields will come afterwards, at least not enough not to justify using an extra byte for PinOptions (which will always be used).